pkg/preferences/userpreferences.go

```diff
@@ -53,7 +54,11 @@ type FeedbackPreferences struct {
 
 func GetLocalPassPhraseFile() (string, error) {
 	if _, err := os.Stat(utils.NitricLocalPassphrasePath()); errors.Is(err, os.ErrNotExist) {
-		err = os.WriteFile(utils.NitricLocalPassphrasePath(), []byte(defaultBackendPassPhrase), os.ModePerm)
+		// In non-CI environments we can generate the file to save a step.
+		// in CI environments this file would typically be lost, so it shouldn't auto-generate.
+		if !output.CI {
```

Comment:
If not having the passphrase in CI will cause an error, maybe this should instead throw the error. That way the user can be directed to the docs instead of experiencing an obscure error down the line.

Reply:
Updated to separate generation from finding the file and lift the decision about whether to generate or not to the individual command that has the context of that environment it's running in.